### ISSUE TITLE
Separate Application into builder and instance

### ISSUE THREAD

**Issue by Pyknic** [speedment/speedment#201]
Currently, the generated Application-class is actually a builder for the Speedment instance and not an application. It would make more sence if the builder was actually a builder and it produced an application.

Consider the following (current) code:
```java
final Speedment speedment = new LibraryApplication()
    .withPassword("my_supersecret_password")
    .build();
```

The generated application class is barely used since it is almost immediatly turned into a Speedment object. That limits the amount of stuff you can do by implementing your own application class since you won't use it after the initialization. Instead, I propose the initialization to look something like this:

```java
final LibraryApplication app = new LibraryApplicationBuilder()
    .withPassword("my_supersecret_password")
    .build();
```

The generated class `LibraryApplicationBuilder` is identical to what we have today, but instead of producing a `Speedment`-instance, it produces a `LibraryApplication` instance that extends `SpeedmentImpl`.

```java
public class LibraryApplication extends SpeedmentImpl {

}
```

It enables the developer to add custom logic to the instance by plugging in custom code generation components.

**Comment by Pyknic** [comment]
We should probably separate it into interface and implementation as well.

**Comment by Pyknic** [comment]
This issue is fixed in the `develop-modules` branch and will be included in the `2.4.0 Forest` release.
